e3nn/nn/gate.py

```diff
@@ -186,15 +207,25 @@ def forward(self, features):
         -------
         `torch.Tensor`
             tensor of shape ``(..., irreps_out.dim)``
-        '''
+        """
         with torch.autograd.profiler.record_function('Gate'):
-            scalars, gates, nonscalars = self.sc(features)
+            scalars, gates, gated = self.sc(features)
 
             scalars = self.act_scalars(scalars)
             if gates.shape[-1]:
                 gates = self.act_gates(gates)
-                nonscalars = self.mul(nonscalars, gates)
-                features = torch.cat([scalars, nonscalars], dim=-1)
+                gated = self.mul(gated, gates)
+                features = torch.cat([scalars, gated], dim=-1)
             else:
                 features = scalars
             return features
+
+    @property
+    def irreps_in(self):
+        """Input representations."""
+        return self._irreps_in
+
+    @property
+    def irreps_out(self):
+        """Output representations (a direct sum of ``irreps_scalars`` and ``irreps_gated``)."""
+        return self._irreps_out
```

Comment:
This is definitely good practice, but we also don't do it for any other module in e3nn... trying to think if there's anything this would break

Probably should actually be done for most of the modules? @mariogeiger  Since none of them support changing their irreps after initialization.

Reply:
I don’t really see the pro and con to be honest. If you both think it’s a good practice then I will make it everywhere

Reply:
Pro is that you can't `mod = ...` and later `mod.irreps_in = "apple"`, but that's fairly minor.

I don't think its worth it to make a special effort for other modules, but good to have it here

Reply:
@Linux-cpp-lisp I am confident that this won't break anything for properties that are initialized on construction. The primary reason I would choose to do this is that it provides us with a section in the documentation for the public-facing attributes, so that users know of their existence.

Reply:
Definitely agree it won't break anything.

Didn't occur to me about documentation, I'm used to just putting an `Attributes` section in the class docstring but maybe that doesn't get good results — obviously this is more correct, but does that approach work at all?

Reply:
The use of `@property` makes the documentation more automated and it formats itself nicely. 
![image](https://user-images.githubusercontent.com/3943761/111821351-8a3ff100-88b0-11eb-8c17-cd79c13e8d8d.png)
https://e3nn--227.org.readthedocs.build/en/227/api/nn/nn_gate.html#e3nn.nn.gate.Gate

Reply:
That looks much nicer :smile: